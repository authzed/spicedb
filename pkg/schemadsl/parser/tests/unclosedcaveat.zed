definition another {}

caveat somecaveat(somecondition uint, somebool bool) {
  somemap{

  
}

definition user {}