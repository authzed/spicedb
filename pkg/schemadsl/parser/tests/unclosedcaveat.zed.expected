NodeTypeFile
  end-rune = 113
  input-source = unclosed caveat test
  start-rune = 0
  child-node =>
    NodeTypeDefinition
      definition-name = another
      end-rune = 20
      input-source = unclosed caveat test
      start-rune = 0
    NodeTypeCaveatDefinition
      caveat-definition-name = somecaveat
      end-rune = 113
      input-source = unclosed caveat test
      start-rune = 23
      caveat-definition-expression =>
        NodeTypeCaveatExpession
          caveat-expression-expressionstr = somemap{

  
}

definition user {}
          end-rune = 113
          input-source = unclosed caveat test
          start-rune = 80
      child-node =>
        NodeTypeError
          end-rune = 113
          error-message = Expected one of: [TokenTypeRightBrace], found: TokenTypeEOF
          input-source = unclosed caveat test
          start-rune = 114
      parameters =>
        NodeTypeCaveatParameter
          caveat-parameter-name = somecondition
          end-rune = 58
          input-source = unclosed caveat test
          start-rune = 41
          caveat-parameter-type =>
            NodeTypeCaveatTypeReference
              end-rune = 58
              input-source = unclosed caveat test
              start-rune = 55
              type-name = uint
        NodeTypeCaveatParameter
          caveat-parameter-name = somebool
          end-rune = 73
          input-source = unclosed caveat test
          start-rune = 61
          caveat-parameter-type =>
            NodeTypeCaveatTypeReference
              end-rune = 73
              input-source = unclosed caveat test
              start-rune = 70
              type-name = bool